core/services/workflows/syncer/workflow_registry.go

```diff
@@ -35,6 +37,40 @@ type Head struct {
 	Timestamp uint64
 }
 
+type workflowAsEvent struct {
+	Data      WorkflowMetadata
+	EventType WorkflowRegistryEventType
+}
+
+func (r workflowAsEvent) GetEventType() WorkflowRegistryEventType {
+	return r.EventType
+}
+
+func (r workflowAsEvent) GetData() any {
+	return r.Data
+}
+
+type GetWorkflowMetadataListByDONParams struct {
+	DonID uint32
+	Start uint64
+	Limit uint64
+}
+
+type GetWorkflowMetadataListByDONReturnVal struct {
+	WorkflowMetadataList []WorkflowMetadata
+}
+
+type WorkflowMetadata struct {
+	WorkflowID   [32]byte
+	Owner        common.Address
```

Comment:
Does it have to be raw bytes? In the lower level APIs, there is a libocr type `Account`: https://github.com/smartcontractkit/libocr/blob/adbe57025f12b9958907bb203acba14360bf8ed2/offchainreporting2plus/types/types.go#L302-L303
It is a human readable string that is not meant to be parsed since the encoding is chain-specific, but can still serve as a comparable key for equality checks, map keys, etc.